src/Symfony/Component/Validator/Constraints/AbstractComparisonValidator.php

```diff
@@ -66,13 +66,10 @@ public function validate(mixed $value, Constraint $constraint)
         // the DateTime constructor:
         // https://php.net/datetime.formats
         if (\is_string($comparedValue) && $value instanceof \DateTimeInterface) {
-            // If $value is immutable, convert the compared value to a DateTimeImmutable too, otherwise use DateTime
-            $dateTimeClass = $value instanceof \DateTimeImmutable ? \DateTimeImmutable::class : \DateTime::class;
```

Comment:
references to `DateTime` in the comment above should probably be updated

Reply:
updated thanks